dynawo/sources/Solvers/SolverSIM/DYNSolverSIM.cpp

```diff
@@ -293,162 +280,61 @@ SolverSIM::calculateIC() {
 }
 
 void SolverSIM::solveStep(double /*tAim*/, double& tNxt) {
-  if (recalculateStep_)
-    solveWithStepRecalculation(tNxt);
-  else
-    solveWithoutStepRecalculation(tNxt);
-}
-
-void SolverSIM::solveWithStepRecalculation(double& tNxt) {
   int counter = 0;
   bool redoStep = false;
-  saveInitialValues();
+  skipAlgebraicResidualsEvaluation_ = false;
 
-  // Save the initial number of events in the timeline in case we need to come back in the past
-  int initialEventsSize = 0;
-  int finalEventsSize = 0;
-  if (timeline_)
-    initialEventsSize = timeline_->getSizeEvents();
+  if (!skipNextNR_)
+    saveContinuousVariables();
 
   do {
-    // If we have more than maxNewtonTry consecutive divergence or root changes, the simulation is stopped
+    // If we have more than maxNewtonTry consecutive divergence, the simulation is stopped
     handleMaximumTries(counter);
 
     // New time step value
     h_ = hNew_;
 
-    // Call the Newton-Raphson solver and analyze the root evolution
-    SolverStatus_t status = solve();
-    skipAlgebraicResidualsEvaluation_ = false;
-
-    switch (status) {
-      /* NON_CONV: the Newton-Raphson algorithm fails to converge
-       *   => The time step is decreased
-       *   => The LU decomposition is forced for the next time steps
-       *   => The initial point for the next time step (h = h/2) is the initial point from the previous accepted time step
-       *   => In case we use the recalculateStep strategy, timeline events must be deleted.
-      */
-      case NON_CONV: {
-        handleDivergence(redoStep);
-        restoreInitialValues();
-        countRestart_ = 0;
-        // Erase timeline messages that could have been added between the previous accepted time step and the non convergence
-        if (timeline_) {
-          finalEventsSize = timeline_->getSizeEvents();
-          if (finalEventsSize != initialEventsSize)
-            timeline_->eraseEvents(finalEventsSize - initialEventsSize);
```

Comment:
This timeline update was removed right?

Reply:
Yes, it is. I keep it in the timeline API anyway because I think it could be a feature required by another solver or the simulation one day so it seemed interesting to keep it for me but it can be change if you have another opinion (like no keeping unused code)